Arg, looks like `docker-py` has some `docker.sock` path assumptions built-in, too:
```
---> Making bundle: test-docker-py (in bundles/1.5.0-dev/test-docker-py)
+++ exec docker --daemon --debug --host unix:///go/src/github.com/docker/docker/bundles/1.5.0-dev/test-docker-py/docker.sock --storage-driver vfs --exec-driver native --pidfile /go/src/github.com/docker/docker/bundles/1.5.0-dev/test-docker-py/docker.pid
..................................E.......................E
======================================================================
ERROR: test_443 (__main__.TestRegressions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/docker-py/tests/integration_test.py", line 1366, in test_443
    for line in self.client.build(fileobj=dfile, tag="a/b/c"):
  File "/docker-py/docker/client.py", line 476, in build
    timeout=timeout,
  File "/docker-py/docker/client.py", line 78, in _post
    return self.post(url, **self._set_request_timeout(kwargs))
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 497, in post
    return self.request('POST', url, data=data, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 455, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 558, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/adapters.py", line 378, in send
    raise ConnectionError(e)
ConnectionError: UnixHTTPConnectionPool(host='localhost', port=None): Max retries exceeded with url: /run/docker.sock/v1.16/build?pull=True&nocache=False&q=False&t=a%2Fb%2Fc&forcerm=False&rm=False (Caused by <class 'socket.error'>: [Errno 2] No such file or directory)

======================================================================
ERROR: test_resource_warnings (__main__.UnixconnTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/docker-py/tests/integration_test.py", line 1347, in test_resource_warnings
    client.images()
  File "/docker-py/docker/client.py", line 648, in images
    res = self._result(self._get(self._url("/images/json"), params=params),
  File "/docker-py/docker/client.py", line 81, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 467, in get
    return self.request('GET', url, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 455, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 558, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/adapters.py", line 378, in send
    raise ConnectionError(e)
ConnectionError: UnixHTTPConnectionPool(host='localhost', port=None): Max retries exceeded with url: /run/docker.sock/v1.16/images/json?only_ids=0&all=0 (Caused by <class 'socket.error'>: [Errno 2] No such file or directory)

----------------------------------------------------------------------
Ran 59 tests in 75.097s

FAILED (errors=2)
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -122,7 +122,7 @@ RUN set -x \
 		go build -o /go/bin/registry-v2 github.com/docker/distribution/cmd/registry
 
 # Get the "docker-py" source so we can run their integration tests
-ENV DOCKER_PY_COMMIT aa19d7b6609c6676e8258f6b900dea2eda1dbe95
+ENV DOCKER_PY_COMMIT d39da1167975aaeb6c423b99621ecda1223477b8
 RUN git clone https://github.com/docker/docker-py.git /docker-py \
 	&& cd /docker-py \
 	&& git checkout -q $DOCKER_PY_COMMIT

--- a/integration-cli/docker_utils.go
+++ b/integration-cli/docker_utils.go
@@ -20,6 +20,8 @@ import (
 	"strings"
 	"testing"
 	"time"
+
+	"github.com/docker/docker/api"
 )
 
 // Daemon represents a Docker daemon for the testing framework.
@@ -266,8 +268,8 @@ func (d *Daemon) Cmd(name string, arg ...string) (string, error) {
 }
 
 func daemonHost() string {
-	daemonUrlStr := "unix:///var/run/docker.sock"
-	if daemonHostVar := os.Getenv("DOCKER_TEST_HOST"); daemonHostVar != "" {
+	daemonUrlStr := "unix://" + api.DEFAULTUNIXSOCKET
+	if daemonHostVar := os.Getenv("DOCKER_HOST"); daemonHostVar != "" {
 		daemonUrlStr = daemonHostVar
 	}
 	return daemonUrlStr
@@ -771,6 +773,12 @@ func fakeGIT(name string, files map[string]string) (*FakeGIT, error) {
 	err = os.Chdir(ctx.Dir)
 	if err != nil {
 		return nil, err
+	}
+	if output, err := exec.Command("git", "config", "user.name", "Fake User").CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("error trying to set 'user.name': %s (%s)", err, output)
+	}
+	if output, err := exec.Command("git", "config", "user.email", "fake.user@example.com").CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("error trying to set 'user.email': %s (%s)", err, output)
 	}
 	if output, err := exec.Command("git", "add", "*").CombinedOutput(); err != nil {
 		return nil, fmt.Errorf("error trying to add files to repo: %s (%s)", err, output)

--- a/project/make.sh
+++ b/project/make.sh
@@ -178,8 +178,21 @@ go_test_dir() {
 		export DEST
 		echo '+ go test' $TESTFLAGS "${DOCKER_PKG}${dir#.}"
 		cd "$dir"
-		go test ${testcover[@]} -ldflags "$LDFLAGS" "${BUILDFLAGS[@]}" $TESTFLAGS
+		test_env go test ${testcover[@]} -ldflags "$LDFLAGS" "${BUILDFLAGS[@]}" $TESTFLAGS
 	)
+}
+test_env() {
+	# use "env -i" to tightly control the environment variables that bleed into the tests
+	env -i \
+		DEST="$DEST" \
+		DOCKER_EXECDRIVER="$DOCKER_EXECDRIVER" \
+		DOCKER_GRAPHDRIVER="$DOCKER_GRAPHDRIVER" \
+		DOCKER_HOST="$DOCKER_HOST" \
+		GOPATH="$GOPATH" \
+		HOME="$DEST/fake-HOME" \
+		PATH="$PATH" \
+		TEST_DOCKERINIT_PATH="$TEST_DOCKERINIT_PATH" \
+		"$@"
 }
 
 # a helper to provide ".exe" when it's appropriate

--- a/project/make/.integration-daemon-start
+++ b/project/make/.integration-daemon-start
@@ -16,8 +16,10 @@ export DOCKER_GRAPHDRIVER=${DOCKER_GRAPHDRIVER:-vfs}
 export DOCKER_EXECDRIVER=${DOCKER_EXECDRIVER:-native}
 
 if [ -z "$DOCKER_TEST_HOST" ]; then
+	export DOCKER_HOST="unix://$(cd "$DEST" && pwd)/docker.sock" # "pwd" tricks to make sure $DEST is an absolute path, not a relative one
 	( set -x; exec \
 		docker --daemon --debug \
+		--host "$DOCKER_HOST" \
 		--storage-driver "$DOCKER_GRAPHDRIVER" \
 		--exec-driver "$DOCKER_EXECDRIVER" \
 		--pidfile "$DEST/docker.pid" \
@@ -26,3 +28,20 @@ if [ -z "$DOCKER_TEST_HOST" ]; then
 else
 	export DOCKER_HOST="$DOCKER_TEST_HOST"
 fi
+
+# give it a second to come up so it's "ready"
+tries=10
+while ! docker version &> /dev/null; do
+	(( tries-- ))
+	if [ $tries -le 0 ]; then
+		if [ -z "$DOCKER_HOST" ]; then
+			echo >&2 "error: daemon failed to start"
+			echo >&2 "  check $DEST/docker.log for details"
+		else
+			echo >&2 "error: daemon at $DOCKER_HOST fails to 'docker version':"
+			docker version >&2 || true
+		fi
+		false
+	fi
+	sleep 2
+done

--- a/project/make/test-docker-py
+++ b/project/make/test-docker-py
@@ -18,8 +18,8 @@ DEST=$1
 			git clone https://github.com/docker/docker-py.git "$dockerPy"
 		}
 
-		export PYTHONPATH="$dockerPy" # import "docker" from our local docker-py
-		python "$dockerPy/tests/integration_test.py"
+		# exporting PYTHONPATH to import "docker" from our local docker-py
+		test_env PYTHONPATH="$dockerPy" python "$dockerPy/tests/integration_test.py"
 	}; then
 		didFail=1
 	fi

--- a/project/make/test-integration-cli
+++ b/project/make/test-integration-cli
@@ -16,9 +16,6 @@ bundle_test_integration_cli() {
 	# even and especially on test failures
 	didFail=
 	if ! {
-		# pull the busybox image before running the tests
-		sleep 2
-
 		source "$(dirname "$BASH_SOURCE")/.ensure-busybox"
 		source "$(dirname "$BASH_SOURCE")/.ensure-emptyfs"
 

--- a/project/make/test-unit
+++ b/project/make/test-unit
@@ -58,7 +58,7 @@ go_run_test_dir() {
 		echo
 		echo '+ go test' $TESTFLAGS "${DOCKER_PKG}${dir#.}"
 		precompiled="$DEST/precompiled/$dir.test$(binary_extension)"
-		if ! ( cd "$dir" && "$precompiled" $TESTFLAGS ); then
+		if ! ( cd "$dir" && test_env "$precompiled" $TESTFLAGS ); then
 			TESTS_FAILED+=("$dir")
 			echo
 			echo "${RED}Tests failed: $dir${TEXTRESET}"

